regulations/static/regulations/js/source/views/main/star-processor.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+var $ = require('jquery');
+
+/***
+ * Functions to process the different types of "stars", indicators that text
+ * can be expanded. Each accepts an $li which may need to be collapsed in
+ * addition to an $expander to expand collapsed $li texts. Each method returns
+ * an updated $expander.
+ ***/
+var StarProcessor = {
```

Comment:
Sounds good. This code was moved around a few times to make it easier to test, but I can definitely clean that up.